src/main/thrift/parquet.thrift

```diff
@@ -190,6 +190,35 @@ enum FieldRepetitionType {
   /** The field is repeated and can contain 0 or more values */
   REPEATED = 2;
 }
+/**
+ * A structure for capturing metadata for estimating the unencoded, uncompressed size
+ * of data.
+ */ 
+struct SizeEstimationStatistics {
+   /** 
+    * The number of logic bytes needed to store present/non-null values.
+    * Unless specified below, the computed size is the size it would take to plain-encode the underlying
+    * physical type.
+    * Special calculations:
+    *  - Enum: plain-encoded BYTE_ARRAY size
+    *  - Integers (same size used for signed and unsigned): int8 - 1 bytes, int16 - 2 
+    *  - Decimal - Each value is assumed to take the minimal number of bytes necessary to encode
+    *    the precision of the decimal value.
+    *  - Nested types (lists, nested groups and maps) - No additional size for these structures
+    *    are accounted for in this field, instead the histogram fields below can be
+    *    be used to estimate overhead to recreate these structures.
+    */
+   1: optional i64 logical_value_byte_storage;
+   /** 
+     * When present there is expected to be one element corresponding to each repetition (i.e. size=max repetition_level+1) 
```

Comment:
@mapleFU @wgtmac one other item, is to decide if we want to store redundant information for max def/rep_level or when do allow exclusion.